### PR TITLE
feat: make the selinux type configurable in the SPOD CRD

### PR DESCRIPTION
**Pull request by ccojocar**
Signed-off-by: Cosmin Cojocar <gcojocar@adobe.com>

#### What type of PR is this?

This extends the SPOD CRD with a field which allow to configure the SELinux type tag used in the security context of SPOd deployment. This is helpful to change the default `spc_t` SELinux type to either a more restrictive type or to `unconfined_t` when SELinux is not enforced. For instance `unconfined_t` can be used with the Flatcar Linux distribution. 

See other solutions proposed in:
https://github.com/kubernetes-sigs/security-profiles-operator/pull/850
https://github.com/kubernetes-sigs/security-profiles-operator/pull/828
https://github.com/kubernetes-sigs/security-profiles-operator/pull/826

#### Does this PR introduce a user-facing change?

```release-note

Add a new field selinuxTypeTag in the SPOD CRD which allows to configure the SELinux type in the SPOd deployment

```

### CODE REVIEW

File: api/spod/v1alpha1/spod_types.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ type SPODSpec struct {
 	// tells the operator whether or not to enable SELinux support for this
 	// SPOD instance.
 	EnableSelinux *bool `json:"enableSelinux,omitempty"`
+	// If specified, the SELinux type tag applied to the security context of SPOD.
+	// +optional
+	SelinuxTypeTag string `json:"selinuxTypeTag,omitempty"`
```
[the remark this answers is not in the document]
Comment (ccojocar) [reply api/spod/v1alpha1/spod_types.go:48]: release not is added, I'll update the docs

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: api/spod/v1alpha1/spod_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccojocar) [reply api/spod/v1alpha1/spod_types.go:48]: I added as well a section into the docs.

File: api/spod/v1alpha1/spod_types.go [hunk 2]
```diff
@@ -43,6 +43,9 @@ type SPODSpec struct {
 	// tells the operator whether or not to enable SELinux support for this
 	// SPOD instance.
 	EnableSelinux *bool `json:"enableSelinux,omitempty"`
+	// If specified, the SELinux type tag applied to the security context of SPOD.
+	// +optional
```
[the remark this answers is not in the document]
Comment (ccojocar) [reply api/spod/v1alpha1/spod_types.go:47]: I added the default value.

**Comment by ccojocar** [comment]
@saschagrunert @JAORMX I fixed all review comments and the tests passed. Please could you have another look? Thanks

**Review by JAORMX — APPROVED** [review]
/lgtm

This is a great addition! thanks for working on this

**Pull request merged**